src/app/shared/scroll-with-loading/scroll-with-loading.directive.ts

```diff
@@ -0,0 +1,29 @@
+import {Directive, EventEmitter, HostListener, Output} from '@angular/core';
+
+export enum LoadDirection {
+    Top = 'top',
+    Bottom = 'bottom',
+}
+
+@Directive({
+    selector: '[appScrollWithLoading]',
+})
+export class ScrollWithLoadingDirective {
+    @Output() loadData = new EventEmitter<LoadDirection>();
+
+    borderOffset = 100;
+    @HostListener('scroll', ['$event.target'])
+    onScroll(target: HTMLElement): void {
+        const scrollPosition = target.scrollTop;
+        const scrollHeight = target.scrollHeight;
+        const clientHeight = target.clientHeight;
```

Comment:
Рекомендую данный подход по объявлению констант из объекта
```suggestion
        const {scrollTop, scrollHeight, clientHeight} = target;
```